### PR TITLE
don't bind to an address if the rest api is disabled.

### LINKED ISSUE #2212: Rest API is enabled regardless of the flag being set
Rest API does not adhere to the `--rest-api-enabled` configuration option. It is enabled regardless.

### PR DESCRIPTION
**Pull request by rolfyone**
- when the rest api is not enabled (rest-api-enabled=false), then i expect that no connection is possible on the specified port.

 fixes #2212

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java [hunk 1]
```diff
@@ -144,17 +145,23 @@ public BeaconRestApi(final DataProvider dataProvider, final TekuConfiguration co
...
       final DataProvider dataProvider, final TekuConfiguration configuration, final Javalin app) {
     this.app = app;
+    this.restApiEnabled = configuration.isRestApiEnabled();
     initialize(dataProvider, configuration);
   }
 
   public void start() {
-    app.start();
+    if (restApiEnabled) {
+      app.start();
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java]: Can we avoid creating `BeaconRestApi` entirely if it's disabled?

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java]: i'll see how ugly it gets, I was trying to minimise the change but maybe.

File: services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java [hunk 2]
```diff
@@ -117,7 +117,7 @@
   private volatile BlockImporter blockImporter;
   private volatile RecentChainData recentChainData;
   private volatile Eth2Network p2pNetwork;
-  private volatile BeaconRestApi beaconRestAPI;
+  private volatile Optional<BeaconRestApi> beaconRestAPI;
```
Comment (ajsutton) [inline services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java]:
```suggestion
  private volatile Optional<BeaconRestApi> beaconRestAPI = Optional.empty();
```

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**